src/printer.js

```diff
@@ -626,7 +628,7 @@ function genericPrintNoParens(path, options, print) {
           concat([
             leftBrace,
             indent(
-              options.tabWidth,
+              options.tabWidth * (parentIsUnionTypeAnnotation ? 2 : 1),
```

Comment:
Awesome! I'll try it and fix the tests too! Yesterday evening I was trying to wrap my head around that indentation thing, playing with different values passed via the corresponding flag.